feat: onboarding add pin extension screen

lujakob opened the pull request:
### Describe the changes you have made in this PR

Add a pin extension screen as last step of onboarding.

### Link this PR to an issue

Fixes #2343

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

<img width="840" alt="Bildschirmfoto 2023-04-13 um 15 41 46" src="https://user-images.githubusercontent.com/886152/231878694-1e2bd87f-fdd8-45ad-a093-fd647aa8403c.png">

### How has this been tested?

Manually

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

lujakob commented:
I added code to detect dark mode browser setting, taken from `utils/index.ts` `getTheme` which should detect a users settings

`window.matchMedia("(prefers-color-scheme: dark)").matches`

I changed my Chrome browser setting to dark mode. The welcome screen appears dark, but the mechanism to detect it as taken from `getTheme` does not work. I wonder why the rest of the screen still appears dark, if the code has no effect.

Anyways, if the code in `getTheme` does not work, I think it's a different issue. What do you think @reneaaron ?

reneaaron commented:
@lujakob You are on :fire:, great job! 

Just tested it and worked fine for me. :muscle: Some minor things I noticed during testing: 

- [ ] The images are quite small (you could add the full size images in the original issue or downsize them to the biggest size they can be displayed)
- [ ] Shadows and/or rounding of the box seems a bit off, could you please double check that with the designs? 

Will continue with the code review probably later today / tomorrow, but looks quite good!

lujakob commented:
> @lujakob You are on 🔥, great job!
> 
> Just tested it and worked fine for me. 💪 Some minor things I noticed during testing:
> 
> * [ ]  The images are quite small (you could add the full size images in the original issue or downsize them to the biggest size they can be displayed)
> * [ ]  Shadows and/or rounding of the box seems a bit off, could you please double check that with the designs?
> 
> Will continue with the code review probably later today / tomorrow, but looks quite good!

* I overlooked the width class on the image when copy/pasting. I increased it now to `w-80`. I found the next size `w-96` too big in proportion to the rest of the content. If you think it is still to small, please test `w-96` and let me know.
* I copy pasted the component from `SetPassword` component which does not have rounded corners and shadow. I applied them to the `PinExtension` and also to `SetPassword` to have it consistent.
* I also added the puzzle icon. I think it fits quite well. (didn't notice it on the current component where it is already used)
* I also added the "Alby" icon on point "3." - what do you think?

reneaaron commented on src/app/screens/Onboard/PinExtension/index.tsx:
```diff
@@ -0,0 +1,87 @@
+    utils.redirectPage("options.html#/discover");
+  };
+
+  const imgStyle: CSSProperties = {
```
Can we use tailwind classes here?

lujakob commented on src/i18n/locales/en/translation.json:
```diff
@@ -33,6 +33,12 @@
+        "title": "Pin your Alby extension",
+        "description": "You're almost there. To use Alby conveniently, remember to pin your extension to the browser toolbar:",
+        "explanation": "1. Click on in the top right corner of your browser<0/>2. Find Alby and click to pin it to the toolbar<0/>3. That's it. To access Alby just click on the Alby icon",
+        "next_btn": "Start buzzin \uD83D\uDC1D with Alby"
```
Inserting the emoji translated to the charcode. I used interpolation, this is even better, as translators don't need to handle the icon at all.

reneaaron replied:
They will see this strange charcode, wouldn't they?

lujakob replied:
It should be changed to `"next_btn": "Start buzzin' {{icon}} with Alby"` now. You don't see it?

reneaaron replied:
Now I do, cool! :sunglasses:

lujakob commented on src/app/screens/Onboard/PinExtension/index.tsx:
```diff
@@ -0,0 +1,67 @@
+          label={t("next_btn")}
+          type="submit"
+          primary
+          className="max-sm:w-full"
```
I removed the `type=submit`, if that's what you meant.

reneaaron replied:
No, i meant the classname :thinking:

lujakob replied:
OK, I removed it also.

lujakob commented:
@reneaaron Thanks for reviewing. I resolved all issues.

reneaaron commented on src/app/utils/index.ts:
```diff
@@ -30,13 +31,23 @@ export function getTheme() {
+    : "light";
+}
+
+export function getBrowserType(defaultType?: BrowserType): BrowserType | null {
```
I feel this param is a bit strange. I'd add the default inside the method (I see no use case where it would make sense to be able to configure this on a call level, I see it as a global default)

```
  // Default to "chrome"
  return "chrome";
```

reneaaron commented on src/app/screens/Onboard/PinExtension/index.tsx:
```diff
@@ -0,0 +1,70 @@
+              // eslint-disable-next-line react/jsx-key
+              <img
+                src="assets/icons/alby_icon_yellow.svg"
+                className="relative -top-1 w-5 inline align-bottom dark:invert"
```
```suggestion
                className="w-5 inline align-bottom"
```

The invert of the logo looks not good, I'd just leave the icon the way it is:

![image](https://user-images.githubusercontent.com/100827540/232829306-ae40ddd0-7cb5-456e-b0fa-487080a496c6.png)

reneaaron reviewed (changes requested):
2 little nitpicks, other than that: :100: 

Thanks also for taking the time to rename some functions and introducing a new `useTheme` hook. :muscle: 

tACK

lujakob commented:
> 2 little nitpicks, other than that: 💯
> 
> Thanks also for taking the time to rename some functions and introducing a new `useTheme` hook. 💪
> 
> tACK

Sounds good. I updated.

reneaaron reviewed (approved):
Just tested another time, added some fixes regarding spacing. Great job! :muscle: 

tACK

lujakob commented:
> Just tested another time, added some fixes regarding spacing. Great job! 💪
> 
> tACK

Thanks for reviewing 🙏

reneaaron commented:
I removed the onboarding card now too, since those are kind of duplicate otherwise. cc @stackingsaunter
